### ISSUE TITLE
Planning the next major version

### ISSUE THREAD

**Issue by gj** [osohq/oso#1703]
We’re planning the next major version of the open source library, and we want to work with you! We have big things in store. If you're interested in providing feedback or being included in an early access program, please fill out [this form](https://docs.google.com/forms/d/e/1FAIpQLScYPnf4PAQt67lfCGWcIge0lmNzr8E2SL0WMSP47GZzYtYGKg/viewform). More to come!

**Comment by kapilt** [comment]
It would be good to see some commitment to the oss project, via activity and engagement with the community, re responses on issue  and PRs. As an example publicly defining the roadmap, perhaps via comments on this or which issues are to be included, instead of feedback links to private forms (aka removing discussion). As is the project appears unhealthy based on activity, and as an adopter of it i'm now wondering about alternatives.

**Comment by samscott89** [comment]
Hey @kapilt! Thanks for the input.

We agree -- the goal of this outreach is to gather input for a public roadmap. We prefer to discuss live so we can get into full details on how people are/want to use Oso. But for folks who would rather comment, they can do that when the roadmap is up.

On the commitment front, that's also the goal of having a public roadmap. We'll share what we plan to work on. Additionally, we have some larger internal changes that we plan to make, with the goal of being able to tackle some of the gnarliest bugs/performance issues that we've seen.

**Comment by jedgresham** [comment]
Seconding @samscott89's comments. 

@kapilt Thats a great call. Can I put you down for, "i want to get involved?"

Let me provide some additional (hopefully) clarifying context. I wasn't at Oso when the library was originally built and released so I'm trying to catch up. Checking out the issues and PR's here is great, but I'd really like to identify some passionate users and work with them more closely to turn that list of n number of issues and PR's into a focused and themed list. What I'm aiming for is community involvement, leveling up my understanding of the issues and doing those things efficiently.

Question: We have both github and the community slack where users are working with us. It's not clear that everyone wants github to be the place where they interact with us about the open source project. Do you have advice about how to create connective tissue between github and slack communities to bring them together?

**Comment by kkirsche** [comment]
I think Slack is a difficult place to focus the team on if you want to get larger corporations and users from larger corporations involved. Many companies restrict or block access to websites like Slack, Dropbox, etc., whether for security or other reasons. I think it's important not to make GitHub-only users second class by focusing primarily on Slack, which I would expect will drive a slightly different user base.

I think taking more of the approach that groups like Python's special interest groups (SIGs) take, which is to use a mix of GitHub and forums/mailing lists, Slack is fine also, etc. is for broad coordination with monthly or quarterly calls where involved or interested users can all jump on Zoom, WebEx, BlueJeans, etc. and coordinate directly, as many of those services are not restricted by large enterprises. These calls should have a clear agenda so that they don't end up going off-topic if the group is large enough.

Personally, the focus on Slack has been a driving factor as to why I have not been able to be more involved with this project

**Comment by dvtkrlbs** [comment]
Hey any update on this?

**Comment by gneray** [comment]
Hey @dvtkrlbs, we're working on this as we speak! Thanks for sharing the bits that would be particularly useful to you. If you're interested, I'll mark you down as a candidate to look at the first alpha when it's ready. Sound good?

**Comment by dvtkrlbs** [comment]
Sounds good

**Comment by aaronleopold** [comment]
Hey all, thanks for all your work on Oso! I filled out the form but wanted to follow up with a post here as well for visibility. 

> Another want to have is feature parity for the Rust language support.

Parroting the above, I think this is very important, as well.

I was also wondering how feasible it would be to add support for the Oso instance to be able to manage some state/context in the Rust SDK? To be clear, I am aware of but not referencing the current context/context facts, although I can't seem to find it availabe in Rust, but I am moreso thinking of the ability to manage a database connection pool, etc. This would allow for a more seamless integration for policies defining relations between resources without running into existential assertion issues, which as it stands seems to be a pain point https://github.com/osohq/oso/issues/1705.

I would envision something like:

`policy.polar`
```polar
resource Organization {
  roles = ["owner", "member"]
}

resource Repository {
  relations = { organization: Organization }
  roles = ["owner", "member"]

  "owner" if "owner" on "organization"
  "member" if "member" on "organization"

  # Etc
}

# This is the bit that would be possible with the state/context, naming/usage of oso.state here not overly important
has_relation(organization: Organization, "organization", repo: Repository) if
  repo.organization(oso.state) = organization
```

`main.rs`

```rust
#[derive(Clone)]
struct MyState {
  db: Arc<Database>,
  // etc
}

let mut oso = Oso::new()
  // State/Context should impl Clone + Send + Sync
  .with_state(MyState { ... });

...

impl Repository {
    pub async fn organization(&self, state: MyState) -> Result<Organization> {
        state.db.do_the_thing().await
    }
}
```

**Comment by gneray** [comment]
Bumping this issue. We're actively planning the next major release and would like to share the initial plan and collect feedback. If you're interested in participating, please drop me a line (my first name at osohq dot com)

**Comment by aaronleopold** [comment]
I just noticed the OS library has been deprecated. So I assume there won't necessarily be a next major version, then?

**Comment by dvtkrlbs** [comment]
Will there be a license change with the next release. Especially the deprecation worries me about a license change.

**Comment by gneray** [comment]
We are considering open sourcing different artifacts (e.g., elements of Oso Cloud), and would expect to license those differently. Libraries, clients, etc. though we expect to continue to license using Apache 2.0 or similar.

**Comment by dvtkrlbs** [comment]
That is great to hear. I am looking forward to it

**Comment by dvtkrlbs** [comment]
Is there any update on this

**Comment by gneray** [comment]
No update yet, but the plan are still as described above!

**Comment by voltechs** [comment]
Anyone else feeling a little strung along here?

**Comment by gneray** [comment]
@voltechs That’s totally fair, and if anyone else is feeling that way, that’s on me. We are still a relatively small team making painful prioritization decisions every month. You have my word that we still plan to do this, but I would totally understand if you need to move on to another approach because I cannot guarantee a specific timeline. If you want to discuss any of this directly, feel free to email me at [my first name] at the Oso domain.

**Comment by gneray** [comment]
I've been reflecting on this. I don't feel great about it for obvious reasons and want to do right by you and the others who are in the same situation. What if we offered you free access to Oso Cloud? As in, forever.

**Comment by dvtkrlbs** [comment]
> I've been reflecting on this. I don't feel great about it for obvious reasons and want to do right by you and the others who are in the same situation. What if we offered you free access to Oso Cloud? As in, forever.

Does this mean the next OSS release plan is scrapped completely?

**Comment by gneray** [comment]
Not at all

**Comment by voltechs** [comment]
@gneray I mean, I generally never turn down free things, especially when it comes with a "forever" license. Sign me up! Also, anything the community can do to help you?

**Comment by dvtkrlbs** [comment]
Yeah I would be interested in a free license as well. Might be useful for some side projects.

**Comment by gneray** [comment]
Sounds good. Sign up for Oso Cloud, then email me, and we'll get you sorted (grahamn at osohq dot com)
